alloptions-limit.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+add_action( 'plugins_loaded', 'wpcom_vip_sanity_check_alloptions' );
+
+function wpcom_vip_sanity_check_alloptions() {
+	if ( defined( 'WP_CLI' ) && WP_CLI ) {
+		return;
+	}
+
+	// Warn should *always* be =< die
+	$alloptions_size_warn  =  800000;
+	$alloptions_size_die   = 1000000; // 1000000 ~ 1MB, too big for memcache
+
+	static $alloptions_size = null; // Avoids repeated cache requests
```

Comment:
I'm not sure we need the static here since this function would only be called once on `plugins_loaded`.